lib/chef/provider/mount.rb

```diff
@@ -58,7 +58,7 @@ def initialize(new_resource, run_context)
       end
 
       action :remount do
-        if current_resource.mounted
+        if current_resource.mounted && remount_require?
           if new_resource.supports[:remount]
             converge_by("remount #{current_resource.device}") do
               remount_fs
```

Comment:
Based upon how the action `remount` is defined already, it does remount the endpoint (to maintain idempotency) **but** after checking if the current system supports a `remount`. 
So issue here seems is for the system you are on, it is reported as unsupported for a remount `new_resource.supports[:remount]`, which is when it tries to do an umount + remount. Also the additional check `remount_required?` is unnecessary. The condition `current_resource.mounted` already does that, which is determined from action `mounted?`(see OS files under `https://github.com/chef/chef/tree/main/lib/chef/provider/mount`) where additional conditions, if needed, will go.

Reply:
Hi @neha-p6, seems this is an option (supports remount: true) which we could provide in an chef remount resource as below :
```
mount tmp do
  fstype 'tmpfs'
  device tmp
  options 'nosuid,nodev,noexec'
  supports remount: true
  action :remount
end
```
and then we get `if new_resource.supports[:remount]` as `true`

Reply:
Hi @NAshwini Correct. The resource expects it to be set to true if recipe is expected to do a remount.
Is it not remounting even after setting it to true, is that what you mean?

Reply:
Yes @neha-p6